src/main/java/frc/robot/subsystems/endEffector/EndEffectorSubsystem.java

```diff
@@ -11,7 +11,7 @@
 
 public class EndEffectorSubsystem extends SubsystemBase {
 
-  EndEffectorIO endEffectorIO;
+  public EndEffectorIO endEffectorIO;
```

Comment:
```suggestion
  EndEffectorIO endEffectorIO;
```

It would make more sense to not use the `endEffectorIO` field, as technically we don't want to access it as this breaks field integrity. As this is used to check for `coralLoaded` (and it's really only used once), it doesn't make sense to just use it for one time.

Instead, it makes more sense for `coralLoaded` would actually be a part of `EndEffectorSubsystem` instead. That's something that the end effector can do, so having a public method to check it would be nice. Here's an example of what I mean below (assume `endEffectorIO` is private):

```java
public bool coralLoaded() {
    return endEffectorIO.coralLoaded();
}
```

to keep the code clean, just put this method after the constructor

Reply:
Huh, I remember Zhide writing this already but it somehow didn't get committed.

Reply:
either it didn't get committed or it wasn't pushed